locator/tablets.cc

```diff
@@ -202,6 +202,7 @@ size_t tablet_metadata::external_memory_usage() const {
 
 class tablet_effective_replication_map : public effective_replication_map {
     table_id _table;
+    tablet_sharder _sharder;
```

Comment:
For this use case, tablet_sharder does not need to be a dht::sharder (and the lookup can be merged with the lookup for the host_id too).

Reply:
Which use case do you have in mind? This is in order to implement erm::get_sharder(), which needs to be uniform for vnodes and tablets, so we need some common abstraction.

Reply:
I look at effective_replication_map as a coordiantor-side thing. I know it's used on the replica too, but IMO it's a mistake.

As a thought exercise, we should be able to have replica-only nodes and coordiantor-only nodes, with the coordinator using data_dictionary and effective_replication_map and the replica nodes using replica:: things. We're very far from such isolation (and views muddle it, because a view makes a replica act as a mini-coordinator), but let's keep it in mind.

So, for coordination, we only need the shard when we look up the host to talk to it. But since we use e_r_m on the replica side too, we might need the sharder.